packages/blade/src/components/Dropdown/_decisions/controlled-dropdown.md

```diff
@@ -0,0 +1,192 @@
+# Controlled Dropdown API
+
+Checkout [Dropdown API Decision](./decisions.md) if you're looking for decisions behind the initial API.
+
+## Problem
+
+### Problem Statement 1: State and City Dropdowns
+
+Imagine a usecase where someone has "State" and "City" dropdown and when you select City, the default "State" automatically gets selected.
+
+Currently there is no ([legal](https://codesandbox.io/s/state-fill-usecase-42u1ev?file=/App.tsx)) way to achieve this because we only support Uncontrolled Dropdown where you can set default selection on first render, but every other selection is handled internally only.
+
+### Problem Statement 2: Reset SelectInput value
+
+If you have a "Reset" button in your form, there isn't way to de-select an already selected item ([legally](https://razorpay.slack.com/archives/C01H13RTF8V/p1681287510398639?thread_ts=1681286843.013449&cid=C01H13RTF8V)). For such usecases as well we need controlled Dropdown
+
+## Solutions
+
+### `onChange` + `isSelected` ✅
```

Comment:
This does seem like the best approach but it will compromise the re-usability of ActionList in cases where it's a clickable.